From 28bf6daff68e8a35d1f39e4cddf5e06dcd1cfdcc Mon Sep 17 00:00:00 2001
From: Mark Stapp <mjs@cisco.com>
Date: Mon, 2 Dec 2024 10:29:46 -0500
Subject: [PATCH 119/120] zebra: include EVPN encap info with recursive
 nexthops

Include EVPN encap info with recursively-resolved nexthops, as
we include other kinds of encap info.

Signed-off-by: Mark Stapp <mjs@cisco.com>
(cherry picked from commit d32483031d1a7063e9e72f023d086b00517ecb1b)
---
 zebra/zebra_nhg.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/zebra/zebra_nhg.c b/zebra/zebra_nhg.c
index ccb354c..bf07116 100644
--- a/zebra/zebra_nhg.c
+++ b/zebra/zebra_nhg.c
@@ -1962,6 +1962,13 @@ static struct nexthop *nexthop_set_resolved(afi_t afi,
 					      nexthop->nh_srv6->seg6_segs->encap_behavior);
 	}
 
+	/* Handle evpn nexthop - capture that info also */
+	if (CHECK_FLAG(newhop->flags, NEXTHOP_FLAG_EVPN)) {
+		resolved_hop->nh_encap_type = newhop->nh_encap_type;
+		memcpy(&(resolved_hop->rmac), &(newhop->rmac), ETH_ALEN);
+		SET_FLAG(resolved_hop->flags, NEXTHOP_FLAG_EVPN);
+	}
+
 	resolved_hop->rparent = nexthop;
 	_nexthop_add(&nexthop->resolved, resolved_hop);
 
-- 
2.56.0.rc1.315.gc6ed9934b7-goog

